File: first_unique_character_in_a_string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        vector<int> mp(26, 0);

        for(auto ch: s) mp[ch - 'a']++;

        for(int i=0; i < s.size(); i++) {
            if(mp[s[i] - 'a'] == 1) return i;
        }
        return -1;
    }
};

// Time Complexity : O(n)
// Space Complexity : O(1)

File: first_unique_character_in_a_string.go
func firstUniqChar(s string) int {
    mp := make([]int, 26)

    for _, ch := range s {
        mp[ch - 'a']++
    }

    for index, ch := range s {
        if mp[ch - 'a'] == 1 {
            return index
        }
    }
    return -1
}

// Time Complexity : O(n)
// Space Complexity : O(1)